File: ms5-analytics/app/routers/health.py
import logging

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import text

from app.database import get_db
from app.models.schemas import HealthResponse

logger = logging.getLogger(__name__)

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
async def health_check(db: AsyncSession = Depends(get_db)):
    """Liveness probe checking database connectivity."""
    db_status = "disconnected"
    try:
        await db.execute(text("SELECT 1"))
        db_status = "connected"
    except Exception as e:
        logger.error(f"Health check: DB connection failed: {e}")

    status = "ok" if db_status == "connected" else "degraded"

    return HealthResponse(status=status, db=db_status)

File: ms5-analytics/src/routes/health.js
const { Router } = require('express');
const { getPool } = require('../database');

const router = Router();

router.get('/health', async (_req, res) => {
  let dbStatus = 'disconnected';
  try {
    const pool = getPool();
    await pool.query('SELECT 1');
    dbStatus = 'connected';
  } catch (err) {
    console.error('[health] DB check failed:', err.message);
  }

  const status = dbStatus === 'connected' ? 'ok' : 'degraded';
  res.json({ status, db: dbStatus });
});

module.exports = router;
